apis/cni/v1alpha1/calicoconfig_webhook.go

```diff
@@ -0,0 +1,87 @@
+// Copyright 2022 VMware, Inc. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package v1alpha1
+
+import (
+	"fmt"
+	"reflect"
+
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/apimachinery/pkg/util/validation/field"
+	ctrl "sigs.k8s.io/controller-runtime"
+	logf "sigs.k8s.io/controller-runtime/pkg/log"
+	"sigs.k8s.io/controller-runtime/pkg/webhook"
+)
+
+// log is for logging in this package.
+var calicoconfiglog = logf.Log.WithName("calicoconfig-resource")
+
+func (r *CalicoConfig) SetupWebhookWithManager(mgr ctrl.Manager) error {
+	return ctrl.NewWebhookManagedBy(mgr).
+		For(r).
+		Complete()
+}
+
+var _ webhook.Defaulter = &CalicoConfig{}
+var _ webhook.Validator = &CalicoConfig{}
+
+// +kubebuilder:webhook:path=/mutate-cni-tanzu-vmware-com-v1alpha1-calicoconfig,mutating=true,failurePolicy=fail,groups=cni.tanzu.vmware.com,resources=calicoconfigs,verbs=create;update,versions=v1alpha1,name=mcalicoconfig.kb.io,admissionReviewVersions=v1,sideEffects=None
+
+// Default implements webhook.Defaulter so a webhook will be registered for the type
+func (r *CalicoConfig) Default() {
```

Comment:
same comment as above, remove default if this is no-op

Reply:
Fixed